src/diffusers/schedulers/deprecated/scheduling_dpmsolver_singlestep_legacy.py

```diff
@@ -0,0 +1,866 @@
+# Copyright 2023 TSAIL Team and The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
```

Comment:
I don't think it makes sense to create a whole new "legacy" scheduler class here. I'd advocate for just deprecating it in the original file